packages/compass-crud/src/components/document.jsx

```diff
@@ -14,7 +14,10 @@ const Document = (props) => {
   } = props;
 
   const doc = useMemo(() => {
-    if (_doc?.isRoot?.()) {
+    // COMPASS-5872 If _doc is a plain js object, it can have isRoot prop,
+    // which would cause the isRoot() to throw an error. So an extended check.
```

Comment:
```suggestion
    // COMPASS-5872 If _doc is a plain js object rather than an instance of hadron-document Document
    // it will not have an isRoot prop, which would cause the isRoot() to throw an error.
```